httpserver-rs/src/lib.rs

```diff
@@ -192,6 +192,19 @@ impl HttpServerCore {
                     let ld = linkdefs.clone();
                     let arc_inner = arc_inner.clone();
                     async move {
+                        if let Some(mode) = ld.values.get("READONLY_MODE"){
+                            match mode.parse() {
+                                Ok(true) => {
+                                    if method!= http::method::Method::GET || method!= http::method::Method::HEAD{
+                                        panic!("Cannot use other methods in Read Only Mode");
```

Comment:
We shouldn't panic and die here. Instead we should still return an `Ok<Response>`, with a status code of... I'd say 400 (bad request)